docs/Coding-Guidelines.md

```diff
@@ -57,3 +57,66 @@ ABI.
 GitHub diff is line oriented. Keeping lines within the preferred limit makes changes easier to
 review. Use a tool like Prettier to bulk-reformat files, or configure your editor's "rulers." If new
 languages become common, we will describe the coding guidelines for such languages here.
+
+## Git Checkin-to-main Policy
+
+The main branch needs to be kept 'healthy' and ready to ship near daily (daily, or within a day or
+two). An unhealthy main branch negatively impacts day-to-day development and jeopardizes release
+schedules. We have policies to ensure a high quality bar in main. The details differ depending if a
+code change is Experimental vs Preview or Stable.
+
+See [Final Decision: Release channels and cadences #637](https://github.com/microsoft/WindowsAppSDK/discussions/637)
+for the expectations of the various release channels.
+
+### Policy: Preview and Stable (aka non-Experimental)
+
+The following requirements must be met to checkin non-Experimental changes to main:
+
+1. API Review approval
+2. Tests
+3. Prepared to accept bug reports from the world
+
+Changes of a non-Experimental nature must meet these policies to be committed to the main branch.
+
+### Policy: Experimental
+
+The policies for Experimental content differ from non-Experimental content because of the reasons for Experimental content:
+
+1. **Selfhost** – a change is good but you're interested in community usage and feedback to verify
+   it's ready for Preview/Stable.
+2. **Skeleton** – API Review has approved and you've made an initial 'skeletal' implementation to
+   get the interface and infrastructure building but isn't functionally complete e.g. skeletal code
+   compiles but everything returns E_NOTIMPL/false/null/etc). Having the API surface building in
+   main can unblock documentation, test and sample writers.
+3. **Feedback** – your API shape isn't locked (or even clear) but you want and/or need feedback from
+   the community, partners and/or others before deciding the API is ready to consider for
+   non-Experimental releases. The API is not yet ready to go through API Review.
+
```

Comment:
4. **Under development** - the API is still under development and is used within WinAppSDK by other components. It will eventually be public to get feedback and be self-hosted.

Reply:
The name "under development" is confusing, as this whole category is for under development APIs. All of the code in this category will eventually be public (or is expected to), and will get feedback and be self-hosted. So I don't think we need a 4th reason?